src/action/action_built.cpp

```diff
@@ -323,6 +323,10 @@ void COrder_Built::AiUnitKilled(CUnit &unit)
 
 static std::size_t FindCFramePercent(const std::vector<CConstructionFrame> &cframes, int percent)
 {
+	if (cframes.empty())
+	{
+		return -1;
```

Comment:
returning `std::optional<std::size_t>` might be an option. that ensures the test for validity.